qemu/tests/hotplug_mem_share_discard_data.py

```diff
@@ -0,0 +1,93 @@
+from avocado.core import exceptions
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import utils_misc
+
+from virttest.utils_numeric import normalize_data_size
+from virttest.utils_test.qemu import MemoryHotplugTest
+from virttest.staging import utils_memory
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Memory share and discard-data hotplug test
+    1) Setup hugepages
+    2) Mount /mnt/kvm_hugepage
+    3) Boot guest with one numa node
+    4) Hotplug 1G memory
+    5) Check memory
+    6) Unplug memory
+    7) Check memory
+    8) Check results for discard-data value
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    timeout = float(params.get("login_timeout", 240))
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+
+    mem_name = params["target_mems"]
+    hp_size = utils_memory.read_from_meminfo("Hugepagesize")
+    hp_free = int(utils_memory.read_from_meminfo('HugePages_Free'))
+    size_target_mem = params["size_mem_%s" % mem_name]
+
+    hp_total = int(utils_memory.read_from_meminfo("HugePages_Total"))
+    hp_plugged = int(normalize_data_size(size_target_mem, "K"))
+    hp_target = int(hp_plugged / hp_size)\
+        + int(hp_total)
+    process.system("echo %s > /proc/sys/vm/nr_hugepages" % hp_target, shell=True)
+
+    is_discard_data = params.get("discard-data_plug1", "yes")
+    is_backend_file = params["backend_mem_plug1"]
+
+    hotplug_test = MemoryHotplugTest(test, params, env)
+    hotplug_test.hotplug_memory(vm, mem_name)
+    hotplug_test.check_memory(vm)
+
+    # Checks if hotplugged device has share: True option
+    if utils_misc.compare_qemu_version(6, 1, 0, is_rhev=False) is True:
+        mem_dev_name = "mem-%s" % params["target_mems"]
+        query_list = vm.monitor.info("memdev")
+        for dev in query_list:
+            error_context.context("device id: %s has share: %s" % (dev['id'], dev['share']), test.log.debug)
+            if dev['id'] == mem_dev_name and dev['share'] is not True:
+                raise exceptions.TestFail("dev %s has share option: %s" % (dev['id'], dev['share']))
```

Comment:
The function `hotplug_memory` include the parameter checking steps already:
https://github.com/avocado-framework/avocado-vt/blob/19b4b6c404049a0e18219a546ddebb82917b9605/virttest/qemu_devices/qdevices.py#L1497

Each option in the output of `info memdev` will be checked, to confirm if it match with the qemu command line param or not.  So i think L54~L61 can be deleted.

Reply:
> Ok, but then, what about the point of checking the `share: True` option only if qemu-kvm >= 6.1.0? The test will only hotplug memory if qemu-kvm >= 6.1.0? 

As long as the `share` in the output of `info memdev`, the function `hotplug_memory` will check it by default. For qemu < 6.1.0, no share option, so no checking.

But yes, one risk it,  `share` should be shown in the meminfo from `info memdev` but it dose not. And the current patch does not cover it too as it use `dev['share']`, not `dev.get('share')`

>You think it doesn't make sense to hotplug memory if we cannot check the share option?

The fact is `Since qemu6.1, memory-backend-ram starts to support share option as well.`, so we test `backend_ram` from qemu6.1.

Reply:
> But yes, one risk it, `share` should be shown in the meminfo from `info memdev` but it dose not. And the current patch does not cover it too as it use `dev['share']`, not `dev.get('share')`

You suggest I should change the `dev['share']` by `dev.get('share')`? And for `dev['id']` need I to do the same?